app/controllers/api/v1/document_change_requests_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Api::V1::DocumentChangeRequestsController < Api::V1::ApplicationController
+  skip_before_action :verify_authenticity_token, only: :update
+  before_action :check_token_and_set_application, only: :update, if: :json_request?
+  before_action :check_file_params_are_present, only: :update, if: :json_request?
+
+  def update
+    @document_change_request = @planning_application.document_change_requests.where(id: params[:id]).first
```

Comment:
`@planning_application.document_change_requests.find(params[:id])` is more rails idiomatic

Reply:
so the reason for using where is that its faster, I remember reading about it whilst working on reducing n+1 queries. Per this stackoverflow using where is preferred from rails 3:
https://stackoverflow.com/questions/9574659/rails-where-vs-find

and this one also advocates find does a lot more:
https://stackoverflow.com/questions/6379099/performance-differences-between-find-and-where-methods


I don't think we should compromise speed over syntax?

Reply:
ok, so instead of find it seems find_by uses "where" under the hood, so we can use that to return a single record and still be efficient https://stackoverflow.com/questions/11161663/find-vs-find-by-vs-where

Reply:
(I really should just read the source code instead of stackoverflow I know, one day)